src/main/kotlin/com/demonwav/mcdev/nbt/tags/TagList.kt

```diff
@@ -0,0 +1,88 @@
+/*
+ * Minecraft Dev for IntelliJ
+ *
+ * https://minecraftdev.org
+ *
+ * Copyright (c) 2017 minecraft-dev
+ *
+ * MIT License
+ */
+
+package com.demonwav.mcdev.nbt.tags
+
+import java.io.DataOutputStream
+import java.util.Objects
+
+class TagList(val type: NbtTypeId, val tags: List<NbtTag>) : NbtTag {
+    // TAG_List has nameless tags, so we don't need to do anything for the names of tags
+    override val payloadSize = 5 + tags.sumBy { it.payloadSize }
+    override val typeId = NbtTypeId.LIST
+
+    override fun write(stream: DataOutputStream) {
+        stream.writeByte(type.typeIdByte.toInt())
+        stream.writeInt(tags.size)
+        tags.forEach { it.write(stream) }
+    }
+
+    override fun equals(other: Any?): Boolean {
+        if (other !is TagList) {
+            return false
+        }
+
+        if (other === this) {
+            return true
+        }
+
+        if (other.type != this.type) {
+            return false
+        }
+
+        if (other.tags.size != this.tags.size) {
+            return false
+        }
+
+        return (0 until this.tags.size).all { i -> other.tags[i] == this.tags[i] }
+    }
+
+    override fun hashCode() = Objects.hash(type, tags.hashCode())
```

Comment:
I'm pretty sure `hash` calls the `hashCode()` method, so `Objects.hash(type, tags)` should be enough